src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -183,13 +209,42 @@ public static PhysicalAddress Parse(string address)
                 validCount++;
             }
 
-            // we too few characters after the last dash
-            if (validCount < 2)
+            // we too few characters after the last delimiter
+            if (validCount < validSegmentLength)
             {
                 throw new FormatException(SR.Format(SR.net_bad_mac_address, address));
             }
 
             return new PhysicalAddress(buffer);
         }
+
+        private static int GetValidSegmentLength(string address, char? delimiter)
```

Comment:
Why does this take `char?` rather than `char`?  If we're calling this, don't we know we have a non-null delimiter?

Reply:
Just because the delimiter defined on 110 is of type `char?`. It can be changed to be just `char`, no problem. I might need to update the calls to it though to make it pass the `char` value instead of the `char?` variable.